src/kratos.ts

```diff
@@ -33,28 +33,26 @@ const fetchKratos = async (
   opts: RequestInit = {},
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
 ): Promise<{ res: any; headers: Headers }> => {
-  return asyncRetry(
-    async () => {
-      const res = await fetch(endpoint, {
-        ...fetchOptions,
-        ...addKratosHeaderCookies(req),
-        ...opts,
-      });
-      if (res.status < 300) {
-        return { res: await res.json(), headers: res.headers };
-      }
-      const err = new KratosError(res.status, await res.text());
-      if (res.status >= 500) {
-        req.log.warn({ err }, 'unexpected error from kratos');
-        throw err;
+  try {
+    const res = await retryFetch(endpoint, {
```

Comment:
Why don't we use `retryFetchParse` here as well?

Reply:
Because we need the response headers. Parse returns only the body parsed as json

Reply:
Ah I see :) 
Missed the headers specifically as I saw you can manually do custom error status on another call.